mmpose/datasets/datasets/top_down/topdown_coco_dataset.py

```diff
@@ -359,14 +371,15 @@ def evaluate(self, outputs, res_folder, metric='mAP', **kwargs):
                 # rescoring
                 n_p['score'] = kpt_score * box_score
 
-            nms = soft_oks_nms if self.soft_nms else oks_nms
-            keep = nms(list(img_kpts), oks_thr, sigmas=self.sigmas)
-
-            if len(keep) == 0:
-                oks_nmsed_kpts.append(img_kpts)
+            if self.use_nms:
+                nms = soft_oks_nms if self.soft_nms else oks_nms
```

Comment:
I'm thinking, whether we could merge `self.soft_nms` and `self.use_nms` to something like `self.nms_method=none|oks|soft`

Reply:
Yes, we can. And this would require deprecation warnings if we were not to change the config files of the models.